src/main/kotlin/org/rust/lang/core/resolve/ref/RsReference.kt

```diff
@@ -17,6 +17,8 @@ interface RsReference : PsiPolyVariantReference {
 
     fun advancedResolve(): BoundElement<RsCompositeElement>? = resolve()?.let { BoundElement(it) }
 
+    fun advancedCachedMultiResolve(): List<BoundElement<RsCompositeElement>> = multiResolve().map { BoundElement(it) }
```

Comment:
API for references is getting uglier and uglier :(

Not sure if we can significantly improve it ...

Reply:
I'm going to move `RsPathReference.resolveInner` to a separate function (like `RsMethodReference`) and then dismantle `advancedResolve()` at all.